src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -1312,11 +1312,20 @@ internal void ThrowExceptionAndWarning(TdsParserStateObject stateObj, bool calle
 
                 if (temp.Count == 1 && temp[0].Exception != null)
                 {
-                    exception = SqlException.CreateException(temp, serverVersion, _connHandler, temp[0].Exception);
+                    exception = SqlException.CreateException(temp, serverVersion, _connHandler, temp[0].Exception, command?.GetBatchCommand(temp[0].BatchIndex));
                 }
                 else
                 {
-                    exception = SqlException.CreateException(temp, serverVersion, _connHandler);
+                    SqlBatchCommand batchCommand = null;
+                    if (temp[0] != null && command != null)
+                    {
+                        int index = temp[0].BatchIndex;
+                        if (temp[0].BatchIndex >= 0)
```

Comment:
```suggestion
                        if (index >= 0)
```

The previous line reads `temp[0].BatchIndex` but it's not used here.

I would suggest newer C# patterns, though (to replace lines 1320-23):

```csharp
if (temp[0]?.BatchIndex is var index and >= 0 && command is not null)
```

Reply:
Seems reasonable. The majority of this implmentation was done before that syntax was available. This branch has been sitting waiting since i first implemented it in /corefx so it's about 4 years old.

I find it interesting that I have to use the `var` syntax, if I try to strongly type index it isn't valid.

Reply:
`if (temp[0]?.BatchIndex is int index and >= 0)` works for me.